service/history/queuev2/virtual_queue.go

```diff
@@ -210,6 +214,14 @@ func (q *virtualQueueImpl) run() {
 }
 
 func (q *virtualQueueImpl) loadAndSubmitTasks() {
+	if err := q.taskLoadRateLimiter.Wait(q.ctx); err != nil {
+		if q.ctx.Err() != nil {
+			return
+		}
+		// this should never happen, but we log it for debugging purposes
+		q.logger.Error("Virtual queue failed to wait for rate limiter", tag.Error(err))
```

Comment:
return after this or let it load tasks?

Reply:
let it load tasks so that the task processing won't be stuck